RoddyCore/src/de/dkfz/roddy/core/CohortDataRuntimeServiceExtension.groovy

```diff
@@ -125,14 +136,94 @@ class CohortDataRuntimeServiceExtension {
 
             def cohortDataSets = superCohort.y.collect { String cohortID, List<String> cohortDatasetIdentifier ->
                 List<DataSet> dList = collectDataSetsForCohort(cohortDatasetIdentifier as String[], analysis, listOfDataSets)
-                dList = dList.sort { a,b -> a.id.compareTo(b.id) }.unique()
+                dList = dList.sort { a, b -> a.id.compareTo(b.id) }.unique()
                 return new CohortDataSet(analysis, cohortID, dList, new File(superCohortOutputDirectory, cohortID))
             }
             return new SuperCohortDataSet(analysis, superCohortID, cohortDataSets, superCohortOutputDirectory)
         }
         return datasets as List<DataSet>
     }
 
+    /**
+     * Returns true, if all filters are valid super cohort identifiers.
+     * @return
+     */
+    boolean checkIfListOnlyContainsSCIdentifiers(List<String> pidFilters) {
+        // We use the COHORT_IDENTIFIER. Rules for it also apply for the super cohort identifier.
+        pidFilters.every { it ==~ COHORT_IDENTIFIER }
+    }
+
+    /**
+     * 1. Check, if super cohort cvalue is present.
+     * 2. Check, if super cohort cvalue is valid.
+     * 3. Check for all cohort identifiers in a super cohort cvalue, if they are present.
+     * 4. Check for all cohort cvalues, if they are valid.
+     * @param configuration
+     * @param scIdentifiers
+     */
+    void validateSuperCohortCValuesOrFail(Configuration configuration, List<String> scIdentifiers) {
+        def values = configuration.configurationValues.allValues
+
+        // 1. Check, if super cohort cvalue is present.
+        def unavailableSuperCohorts = scIdentifiers.findAll() { !configuration.configurationValues.hasValue(it) }
+        if (unavailableSuperCohorts)
+            throw new ConfigurationError((["One or more requested super cohorts are not present in the configuration:"] + unavailableSuperCohorts).join("\n\t"), "Super cohort description missing")
+
+        // 2. Check, if super cohort cvalue is valid.
+        Map<String, String> cohortStrings = scIdentifiers.collectEntries { [it, values[it]?.value] }
+        List<String> invalidCohortListMessages = cohortStrings.collect { String sc, String list ->
+            boolean match = list ==~ "${COHORT_IDENTIFIER}([,]${COHORT_IDENTIFIER}){0,}"
+            return match ? "" : "\t${sc} => ${list}".toString()
+        }.findAll()
+        if (invalidCohortListMessages)
+            throw new ConfigurationError((["One or more cohort strings are invalid:"] + invalidCohortListMessages).join("\n"), "Invalid cohort descriptor")
+
+        // 3. Check for all cohort identifiers in a super cohort cvalue, if they are present.
+        String missingCohorts = scIdentifiers.collect { String sc ->
+            cohortStrings[sc].split(StringConstants.SPLIT_COMMA).collect { String cohortID ->
+                if (!configuration.configurationValues.hasValue(cohortID))
```

Comment:
This is basically again at test, whether its in the configs. Now its called "missing" though. `cohortsNotInConfig`?